wafer/talks/forms.py

```diff
@@ -12,26 +12,37 @@
 
 class TalkForm(forms.ModelForm):
     def __init__(self, *args, **kwargs):
+        self.user = kwargs.pop('user')
         super(TalkForm, self).__init__(*args, **kwargs)
         self.helper = FormHelper(self)
-        submit_button = Submit('submit', _('Submit'))
+
+        actions = (Submit('submit', _('Submit')),)
+
+        if not self.user.has_perm('talks.edit_private_notes'):
+            self.fields.pop('private_notes')
```

Comment:
Okay, even though I actually thought I made it a bit more readable. In the interest of small patches, I've tried your suggestion, it works, so it'll be pushed.